react-app/src/containers/SignUpScreenContainer.js

```diff
@@ -21,9 +22,9 @@ class SignUpScreenContainer extends React.PureComponent {
       const signedUp = await signUpAsync(username, password);
 
       if (signedUp) this.setState({ signedUpModalVisible: true });
-      else { /** TODO: Alert: Username already exists */ }
+      else showAlertModal('Username already exists');
```

Comment:
Is "Username already exists" the only possible reason for this to fail? Couldn't be also something else, like "username too long"?

Reply:
The posible server responses are a 200 or a 500 HTTP status code, therefore what you ask is correct, the server can refuse the account creation for other reason an the message shown to the user will be always _"Username already exists"_.

Taking this into account, there's an utility to verify the username and password before any sign in or sign up request (actually, it's only verify that it has content, but it could be implemented in the way we want).

I think that there are two possibilities here:
- display a generic message or,
- implement the account username and password validation before any request.

@chiiph, what do you think? Neither options have no implementation complexity.

Reply:
Let's use an error message like this: "There was an error creating your account. Try again later, or try with a different username."